core/chains/evm/client/rpc_client.go

```diff
@@ -134,14 +107,11 @@ type rpcClient struct {
 
```

Comment:
This is currently unused. Assuming @dhaidashenko is right that we need to use this to separately protect the subs, we should keep it.  But we might want to change it from `sync.RWMutex` to `sync.Mutex` to be slightly more lightweight.  (Since the only place we read from it is in `UnsubscribeAllExcept`, where we're also writing to it.)